conda/plugins/virtual_packages/cuda.py

```diff
@@ -84,4 +84,6 @@ def cached_cuda_version():
 
 @hookimpl
 def conda_virtual_packages():
-    yield CondaVirtualPackage("cuda", cached_cuda_version())
+    cuda_version = cached_cuda_version()
+    if cuda_version is not None:
+        yield CondaVirtualPackage("cuda", cuda_version, None)
```

Comment:
It's surprising that the virtual plugin API here doesn't take `None` into account as a valid value, which would ignore it during solving.

Reply:
I don't follow...?

Reply:
It's unfortunate that we can't use `None` indicating the unavailability of the virtual package and have to check for it before yielding the virtual package object. I know it's a backward compatibility issue, but still. Nothing to do there FWIW.

Reply:
The API for plugins is still fresh, so I don't know if it's set in stone already to worry too much about backward compatibility. Then again, I also haven't thought about what the best way to communicate absence is; possibly a more structured way to communicate it could be need in future (but IDK).
But well, things to worry about for a release after `22.11.1` ;)